packages/liferay-npm-scripts/src/scripts/storybook.js

```diff
@@ -0,0 +1,148 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+const fs = require('fs');
+const os = require('os');
+const path = require('path');
+
+const getMergedConfig = require('../utils/getMergedConfig');
+const log = require('../utils/log');
+const spawnSync = require('../utils/spawnSync');
+
+const BABEL_CONFIG = getMergedConfig('babel');
+const STORYBOOK_CONFIG = getMergedConfig('npmscripts').storybook;
+
+const NODE_PATHS = [
+	path.join(__dirname, '../../../../node_modules'),
+	path.join(__dirname, '../../node_modules')
+];
+
+const STORYBOOK_CONFIG_DIR_PATH = path.join(__dirname, '../storybook');
+
+const STORYBOOK_CONFIG_FILES = [
+	'addons.js',
+	'config.js',
+	'middleware.js',
+	'preview-head.html',
+	'webpack.config.js'
+];
+
+const PORTAL_ROOT = process.cwd().split('/modules')[0];
+
+/**
+ * Builds the string used for defining the NODE_PATH environment.
+ * On Windows, NODE_PATH is delimited by semicolons (;) instead of colons.
+ * @returns {String}
+ */
+function buildNodePath() {
+	return NODE_PATHS.join(process.platform === 'win32' ? ';' : ':');
+}
+
+/**
+ * Copies storybook config files into the build path.
+ * @param {String} buildPath Path to the build directory.
+ * @param {Array} files The list of files to copy.
+ */
+function copyStorybookConfigFiles(buildPath, files) {
+	files.forEach(function(file) {
+		fs.copyFileSync(
+			path.join(STORYBOOK_CONFIG_DIR_PATH, file),
+			path.join(buildPath, file)
+		);
+	});
+}
+
+/**
+ * Combine multiple Language.properties files to a single file for
+ * liferay-lang-key-dev-loader to read from.
+ * @param {String} buildPath Path to the build directory.
+ * @param {Array} paths List of path strings to Language.properties files.
+ */
+function compileLanguageProperties(buildPath, paths) {
+	const PORTAL_LANG_PATH = path.join(
+		PORTAL_ROOT,
+		'/portal-impl/src/content/Language.properties'
+	);
+
+	const LANG_PATHS = [PORTAL_LANG_PATH, ...paths];
+
+	// Used for keeping track of the current file being processed to show the
+	// correct file that caused an error.
+	let currentLangPath;
+
+	try {
+		const bufferArray = LANG_PATHS.filter(path => fs.existsSync(path)).map(
```

Comment:
Probably shouldn't shadow the outer `path` binding here. I'd make this `langPath` or similar, or better yet, skip the variable entirely and do `LANG_PATHS.filter(fs.existsSync)`. You'd still need to do `.map(langPath => { ... }` though.